lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -1143,37 +1143,122 @@ bool SinkReshapes::run(Function *F, const CompilationContext &cctx) {
     auto *node = &N;
 
     // Sink Reshape below eltwise nodes.
-    if (!node->isDataParallel() || node->hasSideEffects()) {
-      continue;
-    }
+    if (node->isDataParallel() && !node->hasSideEffects()) {
```

Comment:
Why can't we leave this as an early continue? Looks like there's no logic after the `if`.